### PR TITLE
fix: VM-225 Readme update

### PR DESCRIPTION
**Pull request by raftedproc**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,87 +1,153 @@
...
-void store(void *ptr, char value);
-
-/**
- * Returns one byte by a given address in the module memory.
- *
- * @param ptr a address at which the needed byte is located
- * @return the byte at the given address
- */
-char load(void *ptr);
+The Sqlite Wasm module exports a set of Sqlite C API functions.  There are multiple ways to try these Wasm module functions.
```
Comment (mikevoronov) [inline Readme.md:38]: suggest to write SQLite wherever it's possible

Comment (mikevoronov) [reply Readme.md:38]: What about the following variant:

The SQLite Wasm module exports a set of SQLite C API functions. The easiest way to try is to run it with [Marine](https://github.com/fluencelabs/marine) Wasm runtime and Marine [REPL](https://fluence.dev/docs/marine-book/marine-tooling-reference/marine-repl).

File: Readme.md [hunk 2]
```diff
@@ -1,87 +1,153 @@
...
+The fastest way tp try is to run it using a combination of marine and mrepl Rust crates.
+```bash
+/git/sqlite-wasm-connector (master)$cargo install marine mrepl
```
Comment (mikevoronov) [inline Readme.md:41]: `/git/sqlite-wasm-connector (master)` looks too wordy, how about just `$` for bash welcome line.

File: Readme.md [hunk 3]
```diff
@@ -1,87 +1,153 @@
...
+...
+/git/sqlite-wasm-connector (master)$marine repl Config.toml
+```
+You can find MREPL output for a simple "CREATE-INSERT-SELECT" scenario below. Mind the second argument to sqlite3_open_v2() that is an OR-ed Sqlite flag values. 6 in this context is SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE.
```
Comment (mikevoronov) [inline Readme.md:45]: Let's wrap with backticks ` SQLITE_OPEN_READWRITE and SQLITE_OPEN_CREATE

File: Readme.md [hunk 4]
```diff
@@ -1,87 +1,153 @@
...
+result: 100
+ elapsed time: 124.122µs
+
+6> call sqlite3 sqlite3_column_int64 [244584,0]
+result: 42
+ elapsed time: 72.483µs
+
+```
+
+You can try the Sqlite Wasm module using Rust together and [this Sqlite connector](https://github.com/fluencelabs/sqlite-wasm-connector). [Here](https://github.com/fluencelabs/examples/tree/main/marine-examples/sqlite) you can find a simple SQL REPL utility exampl built into a Wasm module. Clone the repo, navigate to the marine-examples/sqlite and run build.sh to bootstrap the SQL REPL example.
```
Comment (mikevoronov) [inline Readme.md:87]: Check grammar here

File: Readme.md [hunk 5]
```diff
@@ -1,87 +1,153 @@
...
+```shell
+/git/fluence_examples/marine-examples/sqlite (master)$./build.sh
+/git/fluence_examples/marine-examples/sqlite (master)$marine repl ./Config.toml
```
Comment (mikevoronov) [inline Readme.md:90]: `mrepl --quite Config.toml`

`--quite` skips mrepl welcome message

File: Readme.md [hunk 6]
```diff
@@ -1,87 +1,153 @@
...
+                    AND B.id=28269 ORDER BY 1;
+SELECT A.id FROM demo_index AS A, demo_index AS B WHERE A.maxX>=B.minX AND A.minX<=B.maxX AND A.maxY>=B.minY AND A.minY<=B.maxY AND B.id=28269 ORDER BY 1
+28215 28216 28262 28269
+
+SQL> quit;
+quit
+result: null
+ elapsed time: 33.303383112s
+
+2>
```
Comment (mikevoronov) [inline Readme.md:137]: let's remove these lines

**Comment by mikevoronov** [comment]
Please be sure that PR header follows conventional commit [style](https://www.conventionalcommits.org/en/v1.0.0/)

File: Readme.md (hunk 2 above)
Comment (raftedproc) [reply Readme.md:46]: Good point

File: Readme.md (hunk 1 above)
Comment (raftedproc) [reply Readme.md]: Done

File: Readme.md (hunk 3 above)
Comment (raftedproc) [reply Readme.md]: Done

File: Readme.md [hunk 7]
```diff
@@ -1,87 +1,153 @@
...
- * @param ptr a address at which the needed byte is located
- * @return the byte at the given address
- */
-char load(void *ptr);
+The Sqlite Wasm module exports a set of Sqlite C API functions.  There are multiple ways to try these Wasm module functions.
+The fastest way tp try is to run it using a combination of marine and mrepl Rust crates.
+```bash
+/git/sqlite-wasm-connector (master)$cargo install marine mrepl
+...
+/git/sqlite-wasm-connector (master)$marine repl Config.toml
```
[the remark this answers is not in the document]
Comment (raftedproc) [reply Readme.md:48]: Done

File: Readme.md (hunk 4 above)
Comment (raftedproc) [reply Readme.md]: Fixed.

File: Readme.md (hunk 5 above)
Comment (raftedproc) [reply Readme.md:104]: Done

File: Readme.md [hunk 8]
```diff
@@ -1,87 +1,153 @@
...
 
-There are three primary sources of nondeterminism in WebAssembly: external functions invocation, NaN payloads, and VM resource exhaustion. The first one is the most problematic; currently, we deal with it by simply prohibiting any interactions with the host environment from Wasm programs. This means that submitted WebAssembly code should not contain any external imports.
+## Contributing
 
-The main purpose of this fork is to compile to Wasm without any imports. SQlite is famous for it embeddability and allows to configure itself by variety of macros. E.g. multithreading that produces a lot of imports of host functions could be completely disabled by setting SQLITE_THREADSAFE to 0. Also this fork has been patched to use only memory database without any interaction with a hard disk. So, now it is Wasm in-memory database.
+Any interested person is welcome to contribute to the project. Please, make sure you read and follow some basic [rules](./CONTRIBUTING.md).
```
Comment (boneyard93501) [inline Readme.md:148]: > [rules](./CONTRIBUTING.md)
dead link

File: Readme.md (hunk 6 above)
Comment (boneyard93501) [reply Readme.md]: also, the Config.toml is outdated. remove: *mem_pages_count = 1000*

File: Readme.md [hunk 9]
```diff
@@ -1,87 +1,153 @@
...
- * Returns one byte by a given address in the module memory.
- *
- * @param ptr a address at which the needed byte is located
- * @return the byte at the given address
- */
-char load(void *ptr);
+
+
+You can also try the SQLite Wasm module using Rust together with [this Sqlite connector](https://github.com/fluencelabs/sqlite-wasm-connector). [Here](https://github.com/fluencelabs/examples/tree/main/marine-examples/sqlite) you can find a simple SQL REPL utility example built into a Wasm module. Clone the repo, navigate to the marine-examples/sqlite and run build.sh to bootstrap the SQL REPL example.
+
```
[the remark this answers is not in the document]
Comment (raftedproc) [reply Readme.md:94]: This contradicts with @mikevoronov intentions so I prefer to left this part.

File: Readme.md [hunk 10]
```diff
@@ -1,87 +1,153 @@
...
-The main purpose of this fork is to compile to Wasm without any imports. SQlite is famous for it embeddability and allows to configure itself by variety of macros. E.g. multithreading that produces a lot of imports of host functions could be completely disabled by setting SQLITE_THREADSAFE to 0. Also this fork has been patched to use only memory database without any interaction with a hard disk. So, now it is Wasm in-memory database.
+Any interested person is welcome to contribute to the project. Please, make sure you read and follow some basic [rules](./CONTRIBUTING.md).
 
-More information about ways of porting C/C++ code to Wasm could be found in our [article](https://medium.com/fluence-network/porting-redis-to-webassembly-with-clang-wasi-af99b264ca8) about porting of Redis.
 
-## Future plans
+## License
 
-We are working on supporting of subset of WASI syscalls and interface types support for more convinient parameter passing scheme.
+All software code is copyright (c) Fluence Labs, Inc. under the [Apache-2.0](./LICENSE) license.
```
[the remark this answers is not in the document]
Comment (raftedproc) [reply Readme.md:153]: It is a common part that I was told by @mikevoronov to borrow from examples.

File: Readme.md (hunk 8 above)
Comment (raftedproc) [reply Readme.md:148]: I would say not yet alive :) 
Nice catch though. I will borrow the file from the other repo where I borrowed the text.

File: Readme.md [hunk 11]
```diff
@@ -1,87 +1,153 @@
...
 ```bash
+
 docker-compose up
+
 ```
 
-or by Makefile with [wasi-sdk](https://github.com/WebAssembly/wasi-sdk) installed
+
+
+or using GNU make. There are prerequisites to be installed following this path, namely [wasi-sdk](https://github.com/WebAssembly/wasi-sdk) and marine crate. Take a look at the Dockerfile for the details.
```
[the remark this answers is not in the document]
Comment (raftedproc) [reply Readme.md:28]: Do you mean adding a link behind the make word?

File: Readme.md (hunk 6 above)
Comment (raftedproc) [reply Readme.md]: Good point.

File: Readme.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (mikevoronov) [reply Readme.md:153]: I am still not sure about licence in this repo, if there are no conflicts with SQLite license and there no any legal issue, so why not. @boneyard93501 wdyt?

File: Readme.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (mikevoronov) [reply Readme.md:94]: I think that @boneyard93501's approach is better because you already made a cool example with raw SQLite API. So suggest moving this example to examples and putting a link here.

File: Readme.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (boneyard93501) [reply Readme.md:153]: i'm pretty you're bound by the upstream license, which is a no-license :) 
https://github.com/sqlite/sqlite/blob/master/LICENSE.md
```
The author disclaims copyright to this source code. In place of a legal notice, here is a blessing:

May you do good and not evil.
May you find forgiveness for yourself and forgive others.
May you share freely, never taking more than you give.
```
sticking apache or mit on it should be ok. but you could just go with the original license.

File: Readme.md [hunk 12]
```diff
@@ -1,87 +1,153 @@
...
+```
+
+
+
+## How to use
+
+
+
+The SQLite Wasm module exports a set of SQLite C API functions. The easiest way to try this module is to run it with Marine Wasm runtime and Marine REPL.
+You can find MREPL output for a simple "CREATE-INSERT-SELECT" scenario below. Mind the second argument to sqlite3_open_v2() that is an OR-ed SQLite flag values. 6 in this context is `SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE`.
```
[the remark this answers is not in the document]
Comment (boneyard93501) [reply Readme.md:44]: don't thim it contradicts anything. you open the repl, you input stuff, the called function is executed and the output is displayed. it just happens that the input stuff is a well though-out sequence of create-insert-select.

File: Readme.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (raftedproc) [reply Readme.md:28]: @boneyard93501 Could you ellaborate on your suggestion?

File: Readme.md (hunk 8 above)
Comment (raftedproc) [reply Readme.md:101]: Fixed

File: Readme.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (raftedproc) [reply Readme.md:106]: Fixed

File: Readme.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (raftedproc) [reply Readme.md]: Fixed

File: Readme.md [hunk 13]
```diff
@@ -1,87 +1,153 @@
...
+
+```
+
+
+
+## How to use
+
+
+
+The SQLite Wasm module exports a set of SQLite C API functions. The easiest way to try this module is to run it with Marine Wasm runtime and Marine REPL.
```
[the remark this answers is not in the document]
Comment (raftedproc) [reply Readme.md]: Fixed

File: Readme.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (boneyard93501) [reply Readme.md:28]: just the link, nothing fancy.

**Review by boneyard93501 — APPROVED** [review]

File: Readme.md [hunk 14]
```diff
@@ -1,87 +1,106 @@
...
 docker-compose up
+
 ```
 
-or by Makefile with [wasi-sdk](https://github.com/WebAssembly/wasi-sdk) installed
+
+
+or using this [Makefile](./Makefile) with GNU make. There are prerequisites to be installed following this path, namely [wasi-sdk](https://github.com/WebAssembly/wasi-sdk) and marine crate. Take a look at the Dockerfile for the details.
+
+
```
Comment (mikhail-1e20) [inline Readme.md:30]: Isn't this obvious after GNU make was mentioned?

File: Readme.md [hunk 15]
```diff
@@ -1,87 +1,106 @@
...
- */
-char load(void *ptr);
+
+
+## How to use
+
+
+
+The SQLite Wasm module exports a set of SQLite C API functions. The easiest way to try this module is to run it with Marine REPL.
+You can find MREPL output for a simple "CREATE-INSERT-SELECT" scenario below. Mind the second argument to sqlite3_open_v2() that is an OR-ed SQLite flag values. 6 in this context is `SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE`.
```
Comment (mikhail-1e20) [inline Readme.md:44]: I would suggest saying smth like "Parameter 6 in this context" since it is usually advised _not_ to begin phrases with numbers. Especially in case the number hasn't appeared yet.

File: Readme.md (hunk 14 above)
Comment (raftedproc) [reply Readme.md:30]: This was a suggestion by @boneyard93501 as I got it.

File: Readme.md (hunk 15 above)
Comment (raftedproc) [reply Readme.md:44]: Good point indeed. I will change '6 in this context' to 'The value "6" in this context'.

File: Readme.md [hunk 16]
```diff
@@ -1,87 +1,106 @@
 # SQLite
 
-Sqlite fork ported to WebAssembly and adapted for the Fluence network. Could be launched and played on the Fluence [dashboard](http://dash.fluence.network/deploy/sqlite).
 
-Based on SQlite version 3.40.1.
+## Overview
+SQLite fork adapted to work with Fluence protocol. The current version is based on SQLite version 3.40.1. The artifact is SQLite WASM module:
```
Comment (mikevoronov) [inline Readme.md:5]:
```suggestion
SQLite fork adapted to work with the Fluence protocol. The current version is based on SQLite version 3.40.1. The artifact is SQLite WASM module:
```

**Review by mikevoronov — APPROVED** [review]

File: Readme.md (hunk 16 above)
Comment (raftedproc) [reply Readme.md]: Done

**Review by mikhail-1e20 — APPROVED** [review]
